src/rpc/net.cpp

```diff
@@ -299,10 +299,10 @@ UniValue getaddednodeinfo(const UniValue& params, bool fHelp)
 
     std::vector<AddedNodeInfo> vInfo = GetAddedNodeInfo();
 
-    if (params.size() == 2) {
+    if (params.size() == 2 || params.size() == 1 && params[0].type() == UniValue::VSTR) {
```

Comment:
Yes, that'd be a possible solution, it would break backwards compat for bitcoin-cli but not for direct JSON-RPC users.
Or we could go all the way, remove backwards compatibility completely and just mention it in the release notes. After all this is a rarely-used call.